From c55225884014a9d37db7433d35edb54d742e506b Mon Sep 17 00:00:00 2001
From: Thomas Wu <quic_wthomas@quicinc.com>
Date: Wed, 9 Oct 2024 13:04:22 -0700
Subject: [PATCH 5/6] wifi: ath12k: Add Power Boost - Compiling

This commit adds symbols, Kconfig, and Makefile changes
that allow to compile Power Boost Feature.

Signed-off-by: Thomas Wu <quic_wthomas@quicinc.com>
---
 drivers/net/wireless/ath/ath12k/Kconfig  | 8 ++++++++
 drivers/net/wireless/ath/ath12k/Makefile | 1 +
 local-symbols                            | 1 +
 3 files changed, 10 insertions(+)

diff --git a/drivers/net/wireless/ath/ath12k/Kconfig b/drivers/net/wireless/ath/ath12k/Kconfig
index 1b91a64..58d8c1a 100644
--- a/drivers/net/wireless/ath/ath12k/Kconfig
+++ b/drivers/net/wireless/ath/ath12k/Kconfig
@@ -105,3 +105,11 @@ config ATH12K_POWER_OPTIMIZATION
 	help
 	  Enable ath12k Power Reduction support
 	  Say Y to enable. If unsure, say N.
+
+config ATH12K_POWER_BOOST
+	bool "ath12k Power Boost support"
+	depends on ATH12K_DEBUGFS
+	depends on RELAY
+	help
+	  Enable ath12k Power Boost support
+	  Say Y to enable. If unsure, say N.
diff --git a/drivers/net/wireless/ath/ath12k/Makefile b/drivers/net/wireless/ath/ath12k/Makefile
index b92e8b7..fc783b8 100644
--- a/drivers/net/wireless/ath/ath12k/Makefile
+++ b/drivers/net/wireless/ath/ath12k/Makefile
@@ -37,6 +37,7 @@ ath12k-$(CPTCFG_ATH12K_AHB) += ahb.o
 ath12k-$(CPTCFG_ATH12K_PPE_DS_SUPPORT) += ppe.o
 ath12k-$(CPTCFG_ATH12K_SAWF) += sawf.o telemetry.o telemetry_agent_if.o
 ath12k-$(CPTCFG_ATH12K_CFR) += cfr.o
+ath12k-$(CPTCFG_ATH12K_POWER_BOOST) += power_boost.o
 
 # for tracing framework to find trace.h
 CFLAGS_trace.o := -I$(src)
diff --git a/local-symbols b/local-symbols
index 556c893..72ea2cb 100644
--- a/local-symbols
+++ b/local-symbols
@@ -192,6 +192,7 @@ ATH12K_CFR=
 ATH12K_MEM_PROFILE_512M=
 ATH12K_SAWF=
 ATH12K_POWER_OPTIMIZATION=
+ATH12K_POWER_BOOST=
 WLAN_VENDOR_ATMEL=
 ATMEL=
 PCI_ATMEL=
-- 
2.25.1

